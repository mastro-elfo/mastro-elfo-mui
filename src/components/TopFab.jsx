import React from "react";
import PropTypes from "prop-types";

import { makeStyles } from "@material-ui/core/styles";
import { Fab, useScrollTrigger, Zoom } from "@material-ui/core";

import KeyboardArrowUpIcon from "@material-ui/icons/KeyboardArrowUp";

const useStyles = makeStyles(theme => ({
  Fab: {
    position: "fixed",
    bottom: theme.spacing(1),
    right: theme.spacing(1),
    zIndex: theme.zIndex.modal
  }
}));

/**
 * A `Fab` that zooms in when the page is scolled and scrolls to top when clicked.
 * @param       {Number} [threshold=100] [description]
 * @param       {[type]} others          [description]
 * @constructor
 */
export default function TopFab({ threshold = 100, ...others }) {
  const classes = useStyles();
  const trigger = useScrollTrigger({ disableHysteresis: true, threshold });

  const handleClick = () => {
    window.scrollTo({ top: 0, left: 0, behavior: "smooth" });
  };

  return (
    <Zoom in={trigger}>
      <Fab onClick={handleClick} className={classes.Fab} {...others}>
        <KeyboardArrowUpIcon />
      </Fab>
    </Zoom>
  );
}

TopFab.propTypes = {
  // Minimum scroll before show the button
  threshold: PropTypes.number
};
