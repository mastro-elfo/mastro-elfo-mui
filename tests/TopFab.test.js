import React from "react";
import renderer from "react-test-renderer";

import TopFab from "../src/components/TopFab";

test("Render", () => {
  const component = renderer.create(<TopFab />);
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
