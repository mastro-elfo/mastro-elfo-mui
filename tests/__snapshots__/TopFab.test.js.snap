// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render 1`] = `
<button
  className="MuiButtonBase-root MuiFab-root makeStyles-Fab-1"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "transform": "scale(0)",
      "visibility": "hidden",
    }
  }
  tabIndex={0}
  type="button"
>
  <span
    className="MuiFab-label"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
      />
    </svg>
  </span>
</button>
`;
